dialog.ahk

```diff
@@ -10,19 +10,40 @@ Gui, Show, CenterAutosize, Update
 Return
 
 Update:
-	request := ComObjCreate("WinHttp.WinHttpRequest.5.1")
+;-------------------------------------------------------------
+; Lexicos says: "WinHttpRequest does not use TLS 1.1 or 1.2 by default on Windows 7 and older".
+; So he uses Msxml2.XMLHTTP here: https://github.com/Lexikos/AutoHotkey-Release/commit/4622287857dbbb2f23dfc91615c5b3b6650241ac
+; And we will use it.
+; Frankly speaking, I get strange encoding in var text while using your WinHttpRequest code. With Msxml2.XMLHTTP everything is ok.
+;-------------------------------------------------------------
+	request := ComObjCreate("Msxml2.XMLHTTP")
 	request.Open("GET", "https://raw.githubusercontent.com/iQuerz/MyAHKScript/main/script.ahk", true)
 	request.Send()
-	request.WaitForResponse()
-	text := request.ResponseText
-	StringLen, length, text
-	scriptContent := SubStr(text, 2, length)
-
-	FileAppend, %scriptContent%, temp.txt
-	FileMove, temp.txt, script.ahk, true
+	while (request.readyState != 4 && A_Index <= 50)    ; We will wait for response 5 seconds at maximum (50*100 ms = 5000 ms).
+		sleep 100
+	if request.readyState != 4 {
+		MsgBox, Connection to the server failed, try to update later.
+		if request.readyState != 4  ; We could  get the response while user was reading msgbox's text.
+			ExitApp
+		else MsgBox, A miracle has just happened, we've got the update!
+	}
+  	text := request.ResponseText
+	;-------------------------------------------------------------
+	; What were you doing here? You've just made "urrentVersion = 1.6" instead of "currentVersion = 1.6"
```

Comment:
First of all, thank you very much. Everything you said here is very helpful and I am really thankful. Hey, nobody knows everything on this world. 
BUT - I wanted to add that I had a really weird issue with WinHttpRequest, as it was returning the whole text with a "?" character in front of the whole file. That's the reason I added the (maybe silly) first character removal part. 
Also, keep in mind that I'm relatively new to AHK and I didn't have anyone else to help me test my code so everything that I pushed on here worked on my PC. That's a requirement before commiting.

Reply:
> BUT - I wanted to add that I had a really weird issue with WinHttpRequest, as it was returning the whole text with a "?" character in front of the whole file. That's the reason I added the (maybe silly) first character removal part. 

Ok, I understand now.

> Also, keep in mind that I'm relatively new to AHK and I didn't have anyone else to help me test my code...

Don't worry, I'm not a programmer at all, just use a little bit Autohotkey for my needs. I told you about things that mostly present in the documentation that I hope you'll learn soon. For example:
https://www.autohotkey.com/docs/commands/URLDownloadToFile.htm#XHR
https://www.autohotkey.com/docs/commands/FileAppend.htm#Remarks
https://www.autohotkey.com/docs/commands/Gui.htm#GuiClose
https://www.autohotkey.com/docs/Hotkeys.htm#combo

> ...so everything that I pushed on here worked on my PC. That's a requirement before commiting.

Didin't catch. You haven't tested these suggestions yet? I've tested.